chain/gen/gen.go

```diff
@@ -276,8 +276,12 @@ func CarWalkFunc(nd format.Node) (out []*format.Link, err error) {
 func (cg *ChainGen) nextBlockProof(ctx context.Context, pts *types.TipSet, m address.Address, round int64) (*types.EPostProof, *types.Ticket, error) {
 	mc := &mca{w: cg.w, sm: cg.sm}
 
-	// TODO: REVIEW: Am I doing this correctly?
-	ticketRand, err := mc.ChainGetRandomness(ctx, pts.Key(), crypto.DomainSeparationTag_TicketProduction, pts.Height(), m.Bytes())
+	buf := new(bytes.Buffer)
+	if err := m.MarshalCBOR(buf); err != nil {
+		return nil, nil, xerrors.Errorf("failed to cbor marshal address: %w", err)
+	}
+
+	ticketRand, err := mc.ChainGetRandomness(ctx, pts.Key(), crypto.DomainSeparationTag_TicketProduction, pts.Height(), buf.Bytes())
```

Comment:
My reading of the spec is that the epoch number here should be `round-SPC_LOOKBACK_TICKET` which differs from this when there are null blocks.  `SPC_LOOKBACK_TICKET = 1` from spec.  This is my best reading of the spec, @sternhenri let us know if this is wrong.

Reply:
Yeah, good catch on that. Fixed